diff --git a/.gitconfig b/.gitconfig
new file mode 100644
--- /dev/null
+++ ./.gitconfig
@@ -0,0 +1,22 @@
+[remote "github"]
+    url = git@github.com:huangyingw/vim-fugitive.git
+    fetch = +refs/heads/*:refs/remotes/github/*
+[remote "tpope"]
+    url = git@github.com:tpope/vim-fugitive.git
+    fetch = +refs/heads/*:refs/remotes/tpope/*
+[push]
+    remote = origin
+    remote = github
+[gsync]
+    remote = tpope
+    target = tpope/master
+[commit]
+    message = n
+[branch "dev"]
+    remote = origin
+    merge = refs/heads/dev
+[branch "RunShell"]
+    remote = origin
+    merge = refs/heads/RunShell
+[gdio]
+    ignore = *.gdio.diff
diff --git a/autoload/fugitive.vim b/autoload/fugitive.vim
--- ./autoload/fugitive.vim
+++ ./autoload/fugitive.vim
@@ -288,7 +288,7 @@ function! s:Map(mode, lhs, rhs, ...) abort
   let maps = []
   let flags = a:0 && type(a:1) == type('') ? a:1 : ''
   let defer = flags =~# '<unique>'
-  let flags = substitute(flags, '<unique>', '', '') . (a:rhs =~# '<Plug>' ? '' : '<script>') . '<nowait>'
+  let flags = substitute(flags, '<unique>', '', '') . (a:rhs =~# '<Plug>' ? '' : '<script>')
   for mode in split(a:mode, '\zs')
     if a:0 <= 1
       call add(maps, mode.'map <buffer>' . substitute(flags, '<unique>', '', '') . ' <Plug>fugitive:' . a:lhs . ' ' . a:rhs)
@@ -6554,6 +6554,10 @@ function! s:IsConflicted() abort
 endfunction
 
 function! fugitive#Diffsplit(autodir, keepfocus, mods, arg, ...) abort
+  only
+  set winwidth=1
+  windo set wrap
+  syntax off
   exe s:VersionCheck()
   let args = s:ArgSplit(a:arg)
   let post = ''
@@ -7073,7 +7077,7 @@ function! s:BlameSubcommand(line1, count, range, bang, mods, options) abort
           let reload = '|call fugitive#DidChange(fugitive#Result(' . string(temp_state.file) . '))'
           let mods = s:Mods(a:mods)
           if a:count != 0
-            exe 'silent keepalt' mods get(a:options, 'curwin') ? 'edit' : 'split' s:fnameescape(temp)
+            exe 'silent keepalt' mods get(a:options, 'curwin') ? 'edit' : 'vsplit' s:fnameescape(temp)
           elseif !&modified || a:bang || &bufhidden ==# 'hide' || (empty(&bufhidden) && &hidden)
             exe 'silent' mods 'edit' . (a:bang ? '! ' : ' ') . s:fnameescape(temp)
           else
@@ -7118,24 +7122,8 @@ function! s:BlameSubcommand(line1, count, range, bang, mods, options) abort
         setlocal scrollbind nowrap nofoldenable
         let top = line('w0') + &scrolloff
         let current = line('.')
-        exe 'silent keepalt' (a:bang ? s:Mods(mods) . 'split' : s:Mods(mods, 'leftabove') . 'vsplit') s:fnameescape(temp)
+        exe 'silent keepalt' s:Mods(mods, 'leftabove') . 'vsplit' s:fnameescape(temp)
         let w:fugitive_leave = join(restore, '|')
-        execute top
-        normal! zt
-        execute current
-        setlocal nonumber scrollbind nowrap foldcolumn=0 nofoldenable winfixwidth
-        if exists('+relativenumber')
-          setlocal norelativenumber
-        endif
-        if exists('+signcolumn')
-          setlocal signcolumn=no
-        endif
-        execute "vertical resize ".(s:linechars('.\{-\}\s\+\d\+\ze)')+1)
-        redraw
-        syncbind
         exe s:DoAutocmdChanged(temp_state)
       endif
     endtry
@@ -7969,7 +7957,7 @@ function! fugitive#MapJumps(...) abort
       endif
 
       call s:Map('n', 'D', ":echoerr 'fugitive: D has been removed in favor of dd'<CR>", '<silent><unique>')
-      call s:Map('n', 'dd', ":<C-U>call fugitive#DiffClose()<Bar>Gdiffsplit!<CR>", '<silent>')
+      call s:Map('n', 'dd', ":<C-U>call fugitive#DiffClose()<Bar>Gvdiffsplit!<CR>", '<silent>')
       call s:Map('n', 'dh', ":<C-U>call fugitive#DiffClose()<Bar>Ghdiffsplit!<CR>", '<silent>')
       call s:Map('n', 'ds', ":<C-U>call fugitive#DiffClose()<Bar>Ghdiffsplit!<CR>", '<silent>')
       call s:Map('n', 'dv', ":<C-U>call fugitive#DiffClose()<Bar>Gvdiffsplit!<CR>", '<silent>')
diff --git a/plugin/fugitive.vim b/plugin/fugitive.vim
--- ./plugin/fugitive.vim
+++ ./plugin/fugitive.vim
@@ -544,9 +544,7 @@ let s:addr_other = has('patch-8.1.560') || has('nvim-0.5.0') ? '-addr=other' : '
 let s:addr_tabs  = has('patch-7.4.542') ? '-addr=tabs' : ''
 let s:addr_wins  = has('patch-7.4.542') ? '-addr=windows' : ''
 
-if exists(':G') != 2
-  command! -bang -nargs=? -range=-1 -complete=customlist,fugitive#Complete G   exe fugitive#Command(<line1>, <count>, +"<range>", <bang>0, "<mods>", <q-args>)
-endif
+
 command! -bang -nargs=? -range=-1 -complete=customlist,fugitive#Complete Git exe fugitive#Command(<line1>, <count>, +"<range>", <bang>0, "<mods>", <q-args>)
 
 if exists(':Gstatus') != 2 && get(g:, 'fugitive_legacy_commands', 0)
@@ -587,7 +585,6 @@ if exists(':Gr') != 2
 endif
 exe 'command! -bar -bang -nargs=* -range=-1                -complete=customlist,fugitive#ReadComplete   Gread    exe fugitive#ReadCommand(<line1>, <count>, +"<range>", <bang>0, "<mods>", <q-args>)'
 
-exe 'command! -bar -bang -nargs=* -complete=customlist,fugitive#EditComplete Gdiffsplit  exe fugitive#Diffsplit(1, <bang>0, "<mods>", <q-args>)'
 exe 'command! -bar -bang -nargs=* -complete=customlist,fugitive#EditComplete Ghdiffsplit exe fugitive#Diffsplit(0, <bang>0, "<mods>", <q-args>)'
 exe 'command! -bar -bang -nargs=* -complete=customlist,fugitive#EditComplete Gvdiffsplit exe fugitive#Diffsplit(0, <bang>0, "vertical <mods>", <q-args>)'
 
@@ -692,9 +689,6 @@ augroup fugitive
         \ if FugitiveIsGitDir(expand('<amatch>:p:h')) |
         \   let b:git_dir = s:Slash(expand('<amatch>:p:h')) |
         \   exe fugitive#BufReadStatus(v:cmdbang) |
-        \   echohl WarningMSG |
-        \   echo "fugitive: Direct editing of .git/" . expand('%:t') . " is deprecated" |
-        \   echohl NONE |
         \ elseif filereadable(expand('<amatch>')) |
         \   silent doautocmd BufReadPre |
         \   keepalt noautocmd read <amatch> |
